lib/srv/desktop/rdp/rdpclient/src/client.rs

```diff
@@ -304,6 +327,93 @@ impl Client {
         })
     }
 
+    async fn handle_clipboard(
+        cgo_handle: CgoHandle,
+        write_stream: &mut RdpWriteStream,
+        x224_processor: Arc<Mutex<X224Processor>>,
+        clipboard: &mut Option<String>,
+        clipboard_function: ClipboardFunction,
+    ) -> ClientResult<()> {
+        match clipboard_function {
+            ClipboardFunction::RequestFormatList => {
+                Client::process_cliprdr(x224_processor, write_stream, |c| c.initiate_copy(&[]))
+                    .await?
+            }
+            ClipboardFunction::Update(data) => {
+                clipboard.replace(data);
+                Client::process_cliprdr(x224_processor, write_stream, |c| {
+                    c.initiate_copy(&[ClipboardFormat::new(ClipboardFormatId::CF_UNICODETEXT)])
+                })
+                .await?;
+            }
+            ClipboardFunction::RemoteCopy(data) => {
+                let format = data
+                    .iter()
+                    .find(|cf| {
+                        cf.id() == ClipboardFormatId::CF_UNICODETEXT
+                            || cf.id() == ClipboardFormatId::CF_TEXT
+                    })
+                    .map(ClipboardFormat::id);
+                if let Some(cf) = format {
+                    Client::process_cliprdr(x224_processor, write_stream, move |c| {
+                        c.initiate_paste(cf)
+                    })
+                    .await?;
+                }
+            }
+            ClipboardFunction::FormatDataRequest(req) => {
+                let data = clipboard
+                    .as_ref()
+                    .filter(|_| req == ClipboardFormatId::CF_UNICODETEXT)
```

Comment:
This is an odd way to use filter, shouldn't an `if` suffice?

Reply:
I've changed this, dunno if it's nicer now, filter above helped with returning the same thing (`new_error`) in couple of cases (data is empty or req != UNICODETEXT), now with matches is much bigger

Reply:
I changed it again, I think it's pretty clean now